diff -uNr libarchive-3.6.2/CMakeLists.txt libarchive-3.6.2_patched/CMakeLists.txt
--- libarchive-3.6.2/CMakeLists.txt	2022-12-09 14:38:31.000000000 +0100
+++ libarchive-3.6.2_patched/CMakeLists.txt	2023-07-11 11:56:23.711822296 +0200
@@ -420,7 +420,11 @@
 # Find ZLIB
 #
 IF(ENABLE_ZLIB)
-  FIND_PACKAGE(ZLIB)
+  FIND_PACKAGE(net.zlib-zlib REQUIRED CONFIG)
+  GET_TARGET_PROPERTY(ZLIB_LIBRARIES net.zlib-zlib::zlibstatic LOCATION)
+  SET(ZLIB_INCLUDE_DIR ${net.zlib-zlib_INCLUDE_DIR})
+  SET(ZLIB_FOUND TRUE)
+  MESSAGE("Found ZLIB: ${ZLIB_LIBRARIES} (found version ${net.zlib-zlib_VERSION})")
 ELSE()
   SET(ZLIB_FOUND FALSE) # Override cached value
 ENDIF()
