diff -uNr libarchive-3.6.1/CMakeLists.txt libarchive-3.6.1_patched/CMakeLists.txt
--- libarchive-3.6.1/CMakeLists.txt	2022-04-08 13:40:26.000000000 +0200
+++ libarchive-3.6.1_patched/CMakeLists.txt	2022-04-19 10:51:05.220994394 +0200
@@ -416,7 +416,11 @@
 # Find ZLIB
 #
 IF(ENABLE_ZLIB)
-  FIND_PACKAGE(ZLIB)
+  FIND_PACKAGE(net.zlib-zlib REQUIRED CONFIG)
+  GET_TARGET_PROPERTY(ZLIB_LIBRARIES net.zlib-zlib::zlibstatic LOCATION)
+  SET(ZLIB_INCLUDE_DIR ${net.zlib-zlib_INCLUDE_DIR})
+  SET(ZLIB_FOUND TRUE)
+  MESSAGE("Found ZLIB: ${ZLIB_LIBRARIES} (found version ${net.zlib-zlib_VERSION})")
 ELSE()
   SET(ZLIB_FOUND FALSE) # Override cached value
 ENDIF()
