diff -uNr libarchive-3.3.1_org/CMakeLists.txt libarchive-3.3.1_patched/CMakeLists.txt
--- libarchive-3.3.1_org/CMakeLists.txt	2017-02-25 20:31:44.000000000 +0100
+++ libarchive-3.3.1_patched/CMakeLists.txt	2017-03-22 11:37:45.809738059 +0100
@@ -386,7 +386,10 @@
 # Find ZLIB
 #
 IF(ENABLE_ZLIB)
-  FIND_PACKAGE(ZLIB)
+  FIND_PACKAGE(net.zlib-zlib REQUIRED CONFIG)
+  GET_TARGET_PROPERTY(ZLIB_LIBRARIES net.zlib-zlib::zlibstatic LOCATION)
+  SET(ZLIB_INCLUDE_DIR net.zlib-zlib_INCLUDE_DIR)
+  MESSAGE("Found ZLIB: ${ZLIB_LIBRARIES} (found version ${net.zlib-zlib_VERSION})")
 ELSE()
   SET(ZLIB_FOUND FALSE) # Override cached value
 ENDIF()
